src/entities/MovieForm/ui/MovieForm.jsx

```diff
@@ -1,106 +1,183 @@
-import React, { useState } from "react";
+import React from "react";
+import { useOutletContext } from "react-router";
 import "./MovieForm.css";
 import { genresList } from "../../../entities/SelectTabs/selectTabsMock";
 import { getMovie } from "../lib/getMovie";
+import { useForm, Controller } from "react-hook-form";
 
-export const MovieForm = ({ initialMovie, onFormSubmit }) => {
-  const [movie, setMovie] = useState(getMovie(initialMovie));
+export const MovieForm = ({ initialMovie }) => {
+  const { movie } = useOutletContext() ?? {initialMovie};
+ 
+  const { handleSubmit, control, reset } = useForm({
+    defaultValues: getMovie(movie),
+  });
 
-  const handleFieldChange = (event) => {
-    setMovie((prevMovie) => ({
-      ...prevMovie,
-      [event.target.name]: event.target.value,
-    }));
+  const onFormSubmit = async (data) => {
+    try {
+      let requestOptions;
+      if (movie) {
+        requestOptions = {
```

Comment:
code duplication, can be moved to a variable with a param `method`, the rest is the same

Reply:
done